test/e2e/util/reqserving/verifypods.go

```diff
@@ -0,0 +1,135 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reqserving
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	supportutil "github.com/openshift/hypershift/support/util"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// VerifyRequestServingPodDistribution verifies that only the correct request serving pods
+// are scheduled on request serving nodes with expected replica counts and proper labeling.
+func VerifyRequestServingPodDistribution(ctx context.Context, hc *hyperv1.HostedCluster) error {
+	cpNamespace := fmt.Sprintf("%s-%s", hc.Namespace, hc.Name)
+	client, err := e2eutil.GetClient()
+	if err != nil {
+		return fmt.Errorf("failed to get cluster client: %w", err)
+	}
+
+	var lastErr error
+	pollCtx, cancel := context.WithTimeout(ctx, 15*time.Minute)
```

Comment:
Nit: This 15-minute timeout is same as ComplexVerificationTimeout that is used elsewhere. 
I will just comment here but there are many places where we use constants like `10 * time.Minute`  or `30 * time.Second` but these are already defined as:
```
	DefaultVerificationTimeout = 10 * time.Minute
	ComplexVerificationTimeout = 15 * time.Minute
	DefaultPollingInterval     = 30 * time.Second
```
So, they could be re-used (in a follow-up PR possibly). Otherwise I don't see a point in defining these constants.